src/databricks/azext_databricks/_params.py

```diff
@@ -26,10 +27,16 @@ def load_arguments(self, _):
         c.argument('custom_public_subnet_name', options_list=['--public-subnet'], arg_group='Custom VNET', help='The name of a new public subnet.')
         c.argument('custom_private_subnet_name', options_list=['--private-subnet'], arg_group='Custom VNET', help='The name of a new private subnet.')
         c.argument('sku_name', options_list=['--sku'], arg_type=get_enum_type(['standard', 'premium', 'trial']), help='The SKU tier name.')
+        c.argument('prepare_encryption', action='store_true', help='Flag to enable the Managed Identity for managed storage account to prepare for CMK encryption.')
 
     with self.argument_context('databricks workspace update') as c:
         c.argument('workspace_name', options_list=['--name', '-n'], id_part='name', help='The name of the workspace.')
         c.argument('tags', tags_type)
+        c.argument('prepare_encryption', action='store_true', help='Flag to enable the Managed Identity for managed storage account to prepare for CMK encryption.')
+        c.argument('encryption_key_source', options_list=['--key-source'], arg_group='Encryption', arg_type=get_enum_type(['Default', 'Microsoft.Keyvault']), validator=validate_encryption_values, help='The encryption key source (provider).')
+        c.argument('encryption_key_name', options_list=['--key-name'], arg_group='Encryption', help='The name of KeyVault key.')
+        c.argument('encryption_key_version', options_list=['--key-version'], arg_group='Encryption', help='The version of KeyVault key.')
+        c.argument('encryption_key_vault', options_list=['--key-vault'], arg_group='Encryption', help='The Uri of KeyVault.')

```

Comment:
You mean add an option, not support uri or name with one option?

The design basically followed the interface of [storage](https://github.com/Azure/azure-cli/blob/06f6ec5e07bf3db0881475cb6204e6bb82c04696/src/azure-cli/azure/cli/command_modules/storage/_params.py#L271-L278) keys.

Reply:
@fengzhou-msft Thanks for the clarification, I think it's better if users are able to just specify the vault name instead of the whole URI. Anyway, aligning with the previous design is not bad.